detectable/src/main/java/com/synopsys/integration/detectable/detectables/bitbake/parse/GraphNodeLabelParser.java

```diff
@@ -15,30 +18,39 @@
 //  virtual:native:/workdir/poky/meta/recipes-support/attr/acl_2.3.1.bb -> GraphNodeLabelDetails.recipeSpec
 public class GraphNodeLabelParser {
     private static final String LABEL_PATH_SEPARATOR = "/";
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
 
-    public String parseVersionFromLabel(String label) throws IntegrationException {
-        GraphNodeLabelDetails labelDetails = parseLabelParts(label);
-        return labelDetails.getVersion();
+    public Optional<String> parseVersionFromLabel(String label) {
+        Optional<GraphNodeLabelDetails> labelDetails = parseLabelParts(label);
+        if (labelDetails.isPresent()) {
+            return Optional.of(labelDetails.get().getVersion());
+        }
+        return Optional.empty();
     }
 
-    public String parseLayerFromLabel(String label, Set<String> knownLayerNames) throws IntegrationException {
-        GraphNodeLabelDetails labelDetails = parseLabelParts(label);
-        String recipeSpec = labelDetails.getRecipeSpec();
+    public Optional<String> parseLayerFromLabel(String label, Set<String> knownLayerNames) {
+        Optional<GraphNodeLabelDetails> labelDetails = parseLabelParts(label);
+        if (!labelDetails.isPresent()) {
+            return Optional.empty();
+        }
+        String recipeSpec = labelDetails.get().getRecipeSpec();
         for (String candidateLayerName : knownLayerNames) {
             String possibleLayerPathSubstring = LABEL_PATH_SEPARATOR + candidateLayerName + LABEL_PATH_SEPARATOR;
             if (recipeSpec.contains(possibleLayerPathSubstring)) {
-                return candidateLayerName;
+                return Optional.of(candidateLayerName);
             }
         }
-        throw new IntegrationException(String.format("Graph Node recipe '%s' does not correspond to any known layer (%s)", label, knownLayerNames));
+        logger.warn("Graph Node recipe '%s' does not correspond to any known layer (%s)", label, knownLayerNames);
```

Comment:
IntelliJ interred this problem. `{}` is the replacement character for logger arguments.
```suggestion
        logger.warn("Graph Node recipe '{}' does not correspond to any known layer ({})", label, knownLayerNames);
```